### DISCUSSION TITLE
How to get ranges of design parameters to achieve robust design?

### DISCUSSION THREAD

**Discussion by haythemchelly** [snl-dakota/dakota#18] [Q&A]
Hi all,

I implemented a MATLAB code, which aims to evaluate the value $y$ of a function $f$ from input design parameters $x_1, x_2, ..., x_n$,
$$y=f(x_1, x_2, ..., x_n).$$
I want to find the ranges of the design parameters such that $y \in  [y^\*-\epsilon, y^\*+\epsilon]$, where $\epsilon$ is an arbitrary small parameter. In other words what are the ranges of the design parameters allowing to achieve a specific output value in a robust way (robust design)?

Do you have any idea on how I could use Dakota in order to solve this problem?

Thank you in advance.

**Comment by jadamstephens** [comment]
Dakota doesn't have a method that can directly solve this problem. We brainstormed a few alternatives.

1. Perform an initial optimization to identify $ y^\ast $ and the minimizer $x^\ast$. Beginning from $x^\ast$ as an initial point, perform two additional (local) optimizations: one in which you maximize $f$ subject to the constraint that $ f < y^\ast + \epsilon$, and other in which you minimize $f$ subject to the constraint that $f > y^\ast - \epsilon$. Although this only shows extreme values of $x$, if $\epsilon$ is small, you might be able to argue that $f$ is monotonic in that region and treat the pair of values of $x$ as bounding a hyperrectangle in design space.
2. A more brute force approach: Find $x^\ast$ as in the first proposal, and then perform random sampling in the region. By examining the marginals and correlations, you might be able to draw some conclusions about the robustness of the optimum.
3. Dakota also has an [importance_sampling](https://dakota.sandia.gov/sites/default/files/docs/6.17.0-release/user-html/usingdakota/reference/method-importance_sampling.html) method that, when using the `adapt_import` or `mm_adapt_import` options, tries to steer sampling more toward the neighborhood of "failure". You can define response levels of $y^\ast - \epsilon$ and $y^\ast + \epsilon$, and Dakota would try to refine its understanding of the locations of those surfaces. Dakota will report probabilities of failure (bounded design variables are treated as being uniformly distributed). You aren't necessarily interested in those probabilities, just the samples themselves, which you could use as in the second proposal.
